metabase/metabase
# fix time series with data gaps stacked bar charts crash when enabling segments labels

Closes https://github.com/metabase/metabase/issues/45717

### Description

Fix time series with data gaps stacked bar charts crash when enabling segments labels.

### How to verify

```
select '2024-06-23 00:00:00' as date, 'a' as value_type,	14080 as value
UNION ALL
select '2024-07-24 00:00:00' as date,	'd' as value_type,	0 as value
```

- Create a stacked bar chart where: `date` is x-axis dimension, `value_type` is breakout and `value` is metric.
- Enable segment labels or all labels
- Ensure it does not crash


### Demo

Before
<img width="1764" alt="Screenshot 2024-09-03 at 7 56 24 PM" src="https://github.com/user-attachments/assets/95511d16-12f2-4059-bfe1-86e928806974">

After
<img width="2542" alt="Screenshot 2024-09-03 at 7 56 45 PM" src="https://github.com/user-attachments/assets/b3cc49a3-3fdb-40c3-89e1-8588916a71bf">

### Checklist
